tools/plugins/com.liferay.ide.sdk.core/src/com/liferay/ide/sdk/core/CreateHelper.java

```diff
@@ -0,0 +1,170 @@
+{
+
+    public static final String ANT_PROGRAM_CONFIG_TYPE_ID = IExternalToolConstants.ID_PROGRAM_LAUNCH_CONFIGURATION_TYPE;
+
+    protected IPath currentCreateFile;
+
+    protected SDK sdk;
+
+    private String[] additionalVMArgs;
+
+    private IProgressMonitor monitor;
+
+    public CreateHelper( SDK sdk )
+    {
+        super( ANT_PROGRAM_CONFIG_TYPE_ID );
+
+        this.sdk = sdk;
+
+        setLaunchSync( true );
+        setLaunchInBackground( true );
+        setLaunchCaptureInConsole( true );
+        setLaunchIsPrivate( true );
+    }
+
+    public CreateHelper( SDK sdk, IProgressMonitor monitor )
+    {
+        this( sdk );
+
+        this.monitor = monitor;
+    }
+
+    public ILaunchConfiguration createLaunchConfiguration(
+        IPath buildFile, String arguments, boolean separateJRE, String workingDir ) throws CoreException
+    {
+        ILaunchConfigurationWorkingCopy launchConfig = super.createLaunchConfiguration();
+
+        launchConfig.setAttribute( IExternalToolConstants.ATTR_LOCATION, buildFile.toOSString() );
+        
+        launchConfig.setAttribute( IExternalToolConstants.ATTR_WORKING_DIRECTORY, workingDir );
+
+        launchConfig.setAttribute( IExternalToolConstants.ATTR_TOOL_ARGUMENTS, arguments );
+
+        launchConfig.setAttribute( DebugPlugin.ATTR_CAPTURE_OUTPUT, true);
+
+        launchConfig.setAttribute( "org.eclipse.debug.ui.ATTR_CAPTURE_IN_FILE",
+            SDKCorePlugin.getDefault().getStateLocation().append( "sdk.log" ).toOSString() );
+
+        if( separateJRE )
+        {
+            launchConfig.setAttribute(
+                IJavaLaunchConfigurationConstants.ATTR_MAIN_TYPE_NAME, IAntLaunchConstants.MAIN_TYPE_NAME );
+            launchConfig.setAttribute( IJavaLaunchConfigurationConstants.ATTR_VM_ARGUMENTS, getVMArgumentsAttr() );
+        }
+
+        return launchConfig;
+    }
+
+    private String getVMArgumentsAttr()
+    {
+        StringBuffer args = new StringBuffer( "-Xmx768m" ); //$NON-NLS-1$
+
+        if( !CoreUtil.isNullOrEmpty( additionalVMArgs ) )
+        {
+            for( String vmArg : additionalVMArgs )
+            {
+                args.append( StringPool.SPACE + vmArg );
+            }
+        }
+
+        return args.toString();
+    }
+
+    public void runTarget( IPath createFile, String arguments, boolean separateJRE, String workingDir )
+        throws CoreException
+    {
+        if( isLaunchRunning() )
+        {
+            throw new IllegalStateException( "Existing launch in progress" );
+        }
+
+        this.currentCreateFile = createFile;
+
+        this.currentCreateFile.toFile().setExecutable( true );
+
+        ILaunchConfiguration launchConfig = createLaunchConfiguration( createFile, arguments, separateJRE, workingDir );
+
+        launch( launchConfig, ILaunchManager.RUN_MODE, monitor );
+
+        this.currentCreateFile = null;
+    }
+
+    protected void addUserEntries( RuntimeClasspathModel model ) throws CoreException
```

Comment:
This looks a lot like code in the ant launcher, do we really need it here?  perhaps we can remove it ok?  because the launch for the external tool shouldn't need these libs.